source/index.ts

```diff
@@ -0,0 +1,576 @@
+			...partialOptions
+		};
+
+		const getPackageData = onetime(() => {
+			const packagePath = pkgUp.sync({cwd: parentDir});
+			// Can't use `require` because of Webpack being annoying:
+			// https://github.com/webpack/webpack/issues/196
+			const packageData = packagePath && JSON.parse(fs.readFileSync(packagePath, 'utf8'));
+
+			return packageData ?? {};
+		});
+
+		if (!options.cwd) {
+			if (!options.projectName) {
+				options.projectName = getPackageData().name;
+			}
+
+			if (!options.projectName) {
+				throw new Error('Project name could not be inferred. Please specify the `projectName` option.');
+			}
+
+			options.cwd = envPaths(options.projectName, {suffix: options.projectSuffix}).config;
+		}
+
+		this.#options = options;
+
+		if (options.schema) {
+			if (typeof options.schema !== 'object') {
+				throw new TypeError('The `schema` option must be an object.');
+			}
+
+			const ajv = new Ajv({
+				allErrors: true,
+				format: 'full',
+				useDefaults: true,
+				errorDataPath: 'property'
+			});
+
+			const schema: JSONSchema = {
+				type: 'object',
+				properties: options.schema
+			};
+
+			this.#validator = ajv.compile(schema);
+
+			for (const [key, value] of Object.entries<JSONSchema>(options.schema)) {
+				if (value?.default) {
+					this.#defaultValues[key as keyof T] = value.default;
+				}
+			}
+		}
+
+		if (options.defaults) {
+			this.#defaultValues = {
+				...this.#defaultValues,
+				...options.defaults
+			};
+		}
+
+		if (options.serialize) {
+			this._serialize = options.serialize;
+		}
+
+		if (options.deserialize) {
+			this._deserialize = options.deserialize;
+		}
+
+		this.events = new EventEmitter();
+		this.#encryptionKey = options.encryptionKey;
+
+		const fileExtension = options.fileExtension ? `.${options.fileExtension}` : '';
+		this.path = path.resolve(options.cwd, `${options.configName ?? 'config'}${fileExtension}`);
+
+		const fileStore = this.store;
+		const store = Object.assign(createPlainObject<T>(), options.defaults, fileStore);
+		this._validate(store);
+
+		try {
+			assert.deepEqual(fileStore, store);
+		} catch {
+			this.store = store;
+		}
+
+		if (options.watch) {
+			this._watch();
+		}
+
+		if (options.migrations) {
+			if (!options.projectVersion) {
+				options.projectVersion = getPackageData().version;
+			}
+
+			if (!options.projectVersion) {
+				throw new Error('Project version could not be inferred. Please specify the `projectVersion` option.');
+			}
+
+			this._migrate(options.migrations, options.projectVersion);
+		}
+	}
+
+	/**
+    Get an item.
+
+	@param key - The key of the item to get.
+	@param defaultValue - The default value if the item does not exist.
+	*/
+	get<Key extends keyof T>(key: Key | string): T[Key] | undefined;
+	get<Key extends keyof T, Default = unknown>(key: Key | string, defaultValue?: Default): T[Key] | Default;
+	get<Key extends keyof T, Default = unknown>(key: Key | string, defaultValue?: Default): T[Key] | Default | undefined {
```

Comment:
`Default` needs to be constrained to be the same type as `Key`. Then you can also just return `Default`.